src/as/en.mangafox/src/helper.ts

```diff
@@ -0,0 +1,189 @@
+    Page,
+    Filter,
+    FilterType,
+    ValueRef,
+} from 'aidoku-as';
+
+export class Parser {
+    parseHomePage(document: Html, listType: boolean): MangaPageResult {
+        let result: Manga[] = [];
+        // two different types of collections styles
+        let ul = (listType) ? 'ul.manga-list-4-list > li' : 'ul.manga-list-1-list > li';
+        let elements = document.select(ul).array();
+
+        for (let i=0; i<elements.length; i++) {
+            let elem    = elements[i];
+            const id    = elem.select('a').attr('href').replace('/manga/', '').replace('/', '');
+            const title = elem.select('a').attr('title')
+            const img   = elem.select('img').attr('src');
+            let manga   = new Manga(id, title);
+            manga.cover_url = img;
+            result.push(manga);
+        }
+        document.close();
+        return new MangaPageResult(result, result.length != 0);
+    }
+
+    getMangaDetails(document: Html, mangaId: string): Manga{
+        const title  = document.select('span.detail-info-right-title-font').text().trim()
+        const author = document.select('p.detail-info-right-say a').text().trim();
+        const desc   = document.select('p.fullcontent').text().trim();
+        const image  = document.select('.detail-info-cover-img').attr('src');
+        const stat   = document.select('.detail-info-right-title-tip').text().trim();
+
+        let manga = new Manga(mangaId, title);
+        manga.author      = author;
+        manga.description = desc;
+        manga.cover_url   = image;
+        manga.status      = MangaStatus.Ongoing
+        manga.url         = `https://fanfox.net/manga/${mangaId}`;
+        if(stat == 'Ongoing')   manga.status = MangaStatus.Ongoing;
+        if(stat == 'Completed') manga.status = MangaStatus.Completed;
+
+        let tags: string[] = [];
+        const genreArr = document.select('.detail-info-right-tag-list a').array();
+        for (let i=0; i<genreArr.length; i++) {
+            tags.push(genreArr[i].text().trim());
+        }
+        manga.categories = tags;
+
+        document.close();
+        return manga
+    }
+
+    getChapterList(document: Html, mangaId: string): Chapter[] {
+        let chapters: Chapter[] = [];
+        const elements  = document.select('div#chapterlist ul li').array();
+        for (let i=0; i<elements.length; i++) {
+            let element = elements[i];
+            const url   = element.select('a').attr('href');
+            const id    = url.replace('/manga/', '').replace('/1.html', '');
+            const spl   = id.split('/');
+
+            let title = '';
+            const titleSplit = element.select('p.title3').text().trim().split('-');
+            if (titleSplit.length >= 2) title = titleSplit.slice(1).join('-');
+
+            let dateString = element.select('.title2').text().trim();
+            let dateObject = ValueRef.string(dateString);
+            let date = dateObject.toDate('MMM d,yyyy');
```

Comment:
This uses the user's current locale, but the site uses US locale regardless. As such, this should be
```ts
dateObject.toDate('MMM d,yyyy', 'en_US');
```

Reply:
Implemented the change